File: convert-sorted-array-to-binary-search-tree/convert-sorted-array-to-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* solution(vector<int>nums, int i, int j){
        if(i>j){
            return NULL;
        }
        int mid=(i+j)/2;
        TreeNode * ptr = new TreeNode(nums[mid]);
        ptr->left=solution(nums,i,mid-1);
        ptr->right=solution(nums,mid+1,j);
        return ptr;
    }
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return solution(nums, 0, nums.size()-1);
    }
};

File: convert-sorted-array-to-binary-search-tree/convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return solution(nums, 0, nums.length -1);
    }

    private TreeNode solution(int[] nums,int i,int j){
        if(i > j){
            return null;
        }
        int mid = (i+j)/2;
        TreeNode head = new TreeNode(nums[mid]);
        head.left = solution(nums, i, mid - 1);
        head.right = solution(nums, mid + 1, j);

        return head;
    }
}